portal/src/it/scala/cromwell/pipeline/controller/AuthControllerItTest.scala

```diff
@@ -0,0 +1,100 @@
+package cromwell.pipeline.controller
+
+import java.util.UUID
+
+import akka.http.scaladsl.model.ContentTypes.`application/json`
+import akka.http.scaladsl.model.{HttpEntity, StatusCodes}
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import com.dimafeng.testcontainers.ForAllTestContainer
+import cromwell.pipeline.BaseItTest
+import cromwell.pipeline.controller.AuthController._
+import cromwell.pipeline.datastorage.dto.auth.{SignInRequest, SignUpRequest}
+import cromwell.pipeline.datastorage.dto.{User, UserId}
+import cromwell.pipeline.utils.StringUtils
+import org.scalatest.compatible.Assertion
+import play.api.libs.json.Json
+
+
+class AuthControllerItTest extends BaseItTest with ScalatestRouteTest with ForAllTestContainer {
+
+  import components.controllerModule.authController
+  import components.datastorageModule._
+
+  private val userPassword = "-Pa$$w0rd-"
+
+  "AuthController" when {
+
+    "signIn" should {
+
+      "return token headers if user exists" in {
+        val newUser = getDummyUser(userPassword)
+        awaitForResult {
+          userRepository.addUser(newUser)
+        }
+
+        val signInRequest = SignInRequest(newUser.email, userPassword)
+        val httpEntity = HttpEntity(`application/json`, Json.stringify(Json.toJson(signInRequest)))
+
+        Post("/auth/signIn", httpEntity) ~> authController.route ~> check {
+          status shouldBe StatusCodes.OK
+          checkAuthTokens
+        }
+      }
+    }
+
+    "signUp" should {
+
+      "return token headers if user was successfully registered" in {
+        val newUser = getDummyUser(userPassword)
+        val signUpRequest = SignUpRequest(newUser.email, userPassword, newUser.firstName, newUser.lastName)
+        val httpEntity = HttpEntity(`application/json`, Json.stringify(Json.toJson(signUpRequest)))
+
+        Post("/auth/signUp", httpEntity) ~> authController.route ~> check {
+          status shouldBe StatusCodes.OK
+          checkAuthTokens
+        }
+      }
+    }
+
+    "refresh" should {
+
+      "return updated token headers if refresh token was valid and active" in {
+        val newUser = getDummyUser(userPassword)
+        awaitForResult {
```

Comment:
You could just use [this](http://doc.scalatest.org/3.0.1-2.12/org/scalatest/concurrent/Futures$FutureConcept.html#futureValue(implicitconfig:Futures.this.PatienceConfig,implicitpos:org.scalactic.source.Position):T). Also take a look at [this](http://doc.scalatest.org/3.0.1-2.12/org/scalatest/concurrent/ScalaFutures.html#whenReady[T,U](future:Futures.this.FutureConcept[T])(fun:T=%3EU)(implicitconfig:Futures.this.PatienceConfig,implicitpos:org.scalactic.source.Position):U).

Reply:
Thank you, the implementation has become easier.